carrent/carrentapp/admin.py

```diff
@@ -1,3 +1,80 @@
+from django import forms
 from django.contrib import admin
+from django.contrib.auth.models import Group
+from django.contrib.auth.admin import UserAdmin as BaseUserAdmin
+from django.contrib.auth.forms import ReadOnlyPasswordHashField
+from django.core.exceptions import ValidationError
+
+from .models import UserCustom
+
+
+class UserCreationForm(forms.ModelForm):
+    """A form for creating new users. Includes all the required
+    fields, plus a repeated password."""
+    password1 = forms.CharField(label='Password', widget=forms.PasswordInput)
+    password2 = forms.CharField(label='Password confirmation', widget=forms.PasswordInput)
+
+    class Meta:
+        model = UserCustom
+        fields = ('email', 'username', 'first_name', 'last_name', 'birthdate', 'addr_city', 'addr_street', 'addr_post_code', 'mobile_nr')
+
+    def clean_password2(self):
+        # Check that the two password entries match
+        password1 = self.cleaned_data.get("password1")
+        password2 = self.cleaned_data.get("password2")
+        if password1 and password2 and password1 != password2:
+            raise ValidationError("Passwords don't match")
+        return password2
+
+    def save(self, commit=True):
+        # Save the provided password in hashed format
+        user = super().save(commit=False)
+        user.set_password(self.cleaned_data["password1"])
+        if commit:
+            user.save()
+        return user
+
+
+class UserChangeForm(forms.ModelForm):
```

Comment:
Tak samo jak wcześniej formularz, powinien być w odpowiednim pliku

Reply:
sugerowałem się przykładem z dokumentacji django 4, choć też zdawało mi się bałaganiarskie umieszzcanie klas dziedziczących po forms w adminie.

"Then, to register this custom user model with Django’s admin, the following code would be required in the app’s admin.py file"

quoted from: https://docs.djangoproject.com/en/4.0/topics/auth/customizing/#a-full-example

Reply:
No tak, trzeba mieć w głowie, że w dokumentacji najczęściej dostaniesz info jak to zrobić i też czasami jest to bardziej przejrzyste do pokazania w jednym pliku dla nich.
No i mają też taki ważny tekścik na początku przykładu 
```
Here is an example of an admin-compliant custom user app
```
Zobacz sobie inne przykłady dla porównania
https://testdriven.io/blog/django-custom-user-model/

Reply:
_Here is an example of an admin-compliant custom user app_

no tak a czy to nie jest częścią naszego zadania?

F001 System musi posiadać możliwość dodawania użytkowników(pracowników)

Reply:
Normalni użytkownicy nie muszą być dodawani przez admina, więc jeśli tak jest na dokumencie to błąd, więc teraz tylko jest dodawanie pracowników, tak? Zwykli użytkownicy będą w osobnym PR?

Reply:
tak... tzn teraz admin tez moze dodac zwyklego uzytkownika jesli nie zaznaczy mu admin lub staff
zrobilem tak zeby inni mogli juz pracowacz jakims userem
Ale TAK docelowy formularz logowania bedzie w innym PR